### PR TITLE
Suite highway environment

### PR DESCRIPTION
**Pull request by Haichao-Zhang**
Added suite for highway environment.

### CODE REVIEW

File: alf/environments/suite_highway.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    import highway_env
+except ImportError:
+    highway_env = None
+
+
+def is_available():
+    return highway_env is not None
+
+
+class VectorizeObservation(gym_wrappers.BaseObservationWrapper):
```
Comment (hnyu) [inline alf/environments/suite_highway.py]: The class name can be better called `FlattenObservation`; missing docstring for this class

File: alf/environments/suite_highway.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+                    "sorted"
+            },
+            "action": {
+                "type": "ContinuousAction"
+            }
+        }
+        env_config = default_env_config
+
+    gym_env.configure(env_config)
+    _ = gym_env.reset()
```
Comment (hnyu) [inline alf/environments/suite_highway.py]: gym_env.reset() without returning

File: alf/environments/suite_highway.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    Returns:
+        An AlfEnvironment instance.
+    """
+    assert environment_name in {
+        "highway-v0", "merge-v0", "roundabout-v0", "intersection-v0",
+        "parking-v0"
+    }, "wrong highway environment name"
+
+    gym_spec = gym.spec(environment_name)
+    gym_env = gym.make(environment_name)
```
Comment (hnyu) [inline alf/environments/suite_highway.py]: Does the env itself use gym.make to wrap with max_steps?

**Review by hnyu — CHANGES_REQUESTED** [review]

File: alf/environments/suite_highway.py (hunk 1 above)
Comment (Haichao-Zhang) [reply alf/environments/suite_highway.py]: Changed class name and added doc string

File: alf/environments/suite_highway.py (hunk 2 above)
Comment (Haichao-Zhang) [reply alf/environments/suite_highway.py]: updated

File: alf/environments/suite_highway.py (hunk 3 above)
Comment (Haichao-Zhang) [reply alf/environments/suite_highway.py]: The original env does not differentiate terminations due to max_steps or others.
Now correctly handle max_steps in a way similar to suite_safety_gym.

Comment (hnyu) [reply alf/environments/suite_highway.py]: I mean by calling "gym.make", it probably already wrap an incorrect time_limit wrapper. Should use "gym_spec.make" instead. 
https://alf.readthedocs.io/en/latest/notes/knowledge_base.html#about-timestep-discount

**Review by hnyu — APPROVED** [review]

File: alf/environments/suite_highway.py (hunk 3 above)
Comment (Haichao-Zhang) [reply alf/environments/suite_highway.py]: Right, I changed to use ``gym.spec.make`` instead of ``gym.make``in the second commit because of this.

**Pull request merged**